runtime.md

```diff
@@ -15,3 +15,36 @@ Runs a process in a container. Can be invoked several times.
 Not sure we need that from oc cli. Process is killed from the outside.
 
 This event needs to be captured by oc to run onstop event handlers.
+
+## Hooks
+Hooks allows one to run code before/after various lifecycle events of the container. The state of the container is passed to the hooks over a well-known fd, so the hooks could get the information they need to do their work.
+
+Hooks are placed under a well known directories under which they are search for and executed. The list of directories that will be looked up are:
+
+```
+    /run/ocf/hooks
+    /usr/lib/ocf/hooks
+```
+
+If a hook returns a non-zero exit code, then an error is returned to the caller and the operation is aborted.
+For e.g., if the pre-start hooks fails, then the container isn't started and an error is returned to the
+caller.
+
+### Pre-start
+The pre-start hooks are called after the container process is started but before the user supplied command is execed. It provides an opportunity to customize the container. In Linux, for e.g., the network namespace could be configured in these hook.
+
+### Post-stop
+The post-stop hook are called after the container process is stopped. Cleanup or debugging could be performed in such a hook.
+
+```
+    "hooks" : {
+        "pre-start": [
+            ["fix-mounts", "arg1", "arg2"],
+            ["setup-network", "eth0"]
+        ],
+        "post-stop": [
```

Comment:
`post-stop` means after the container is stopped. What you are describing sounds more like a `pre-stop` hook. 

> Could it be even better to expose different hooks for multiple types of termination signals/events?

Could you elaborate?